### ISSUE TITLE
Changefeeds inevitably overruns their stack

### ISSUE THREAD

**Issue by TotalKrill** [rethinkdb-rs/reql#42]
So after some testing where i start a changefeed, the stack will always get overrun. 

I am going to guess it is due to fact that the data is never ended, thus, never discarded in the background. 

```
*snip*
    let query = r.db("database")
        .table("changing_table")
        .changes()
        .run::<Change<DataStruct, DataStruct>>( conn ).unwrap();

    let mut wait = query.wait();
    loop {
        let val = wait.next();
        match  val {
 *snip*

```
This is snippet of the code I am using. Is there any other way to discard data once it is moved out of the iterator?

**Comment by TotalKrill** [comment]
My suspicions were correct, I managed to hunt down the stack overrun to the Request.submit() command

**Comment by TotalKrill** [comment]
Addressed in #43
